### ISSUE TITLE
Item link attributes

### ISSUE THREAD

**Issue by razvanaldea89** [MachoThemes/modula-lite#221]
Think some options should be added for item link attributes, so that the user can choose whether or not to show link title and which of the title, captionor alt text should be showm

**Comment by ElectricFeet** [comment]
This ties into a ticket I raised with the helpdesk (I have Pro), regarding the ugly tooltips that show up on browsers and mess with the look of the gallery. I'll add the detail here, so it's in one place.

## Issue: Horrible Tooltips on hover
See example:

<img width="284" alt="PastedGraphic-2" src="https://user-images.githubusercontent.com/6003638/70333093-245a6200-1843-11ea-896d-77953f8fcfa2.png">

On the Modula demo pages, the demos in which captions are enabled also do this. For example, [the “Pufrobo" demo]( https://wp-modula.com/demo/hover-effects/pufrobo/ ) — which allows you to click through to a gallery — shows “Caption Text” when you hover anywhere on the image:

<img width="447" alt="PastedGraphic-3" src="https://user-images.githubusercontent.com/6003638/70333167-49e76b80-1843-11ea-9dc4-b8f91f030d07.png">

### Why tooltips are an issue and title attributes are not needed:

1. They're ugly; especially on a well thought-out gallery.

2. They're not needed for SEO: For example, from the [Yoast](https://yoast.com/image-seo-alt-tag-and-title-tag-optimization/) site:

   "Each image should have an alt text, not just for SEO purposes but also because blind and visually impaired people won’t otherwise know what the image is about, **but a title attribute is not required**. What’s more, most of the time it doesn’t make sense to add it. They are only available to mouse (or other pointing devices) users and the only one case where the title attribute is required for accessibility is on <iframe> and <frame> tags.

   If the information conveyed by the title attribute is relevant, consider making it available somewhere else, in plain text and if it’s not relevant, **consider removing the title attribute entirely**.”

   [my emphasis, and note that the title/caption information is already available on the image in plain text] 

3. **WordPress itself removed tooltips/titles in WordPress 4.2** five years ago, specifically [saying](https://core.trac.wordpress.org/ticket/32095#comment:4):

   "In recent years, we've begun to actively discourage the use of title attributes in links as they are largely useless outside of providing the "hover tooltip" many visual users enjoy, and more importantly, they don't promote good accessibility."

## What I tried
I used the `modula_shortcode_item_data` to switch off the titles/tooltips:
```
add_filter( 'modula_shortcode_item_data', 'remove_link_title_attributes', 20, 3 );
function remove_link_title_attributes( $item_data, $item, $settings ) {

    $item_data['link_attributes']['title'] = '';

    return $item_data;
}
```
But unfortunately this switched off the captions on the lighbox popup, which is unacceptable.

## Conclusion
This brings us back to the issue as originally raised above. 

At first I thought that the attribute that is used for the `title=""` in the link should not be the same as the caption in the lightbox -- which would give us a way to switch the title attribute in the link off, while leaving the caption in the  lightbox switched on.

However, thinking through this some more and re-reading the Yoast/WordPress positions, a better solution would be simply to remove the title attribute from the link altogether, but still allow use of the text for the caption.

**Comment by cristianraiber** [comment]
> a better solution would be simply to remove the title attribute from the link altogether, but still allow use of the text for the caption.

^ Yup, that's a good & valid point.

**Comment by cristianraiber** [comment]
@giucu91 - can we make this happen in our next update, please?

**Comment by giucu91** [comment]
@cristianraiber @ElectricFeet this it's fixed but we can delete the title if we want to use : swipebox or prettyPhoto.

**Comment by ElectricFeet** [comment]
Thanks @giucu91 ! Great that it's fixed. 

> but we can delete the title if we want to use : swipebox or prettyPhoto.

I'm a bit confused: With the fix, the title will also be removed from the other hover effects (such as "hide"). Correct?

**Comment by giucu91** [comment]
@ElectricFeet the tooltip will appear only if you use swipebox or prettyPhoto lightbox no matter what hover effect you're using.

**Comment by ElectricFeet** [comment]
Hi there, sorry for not responding before now. Thanks for the fast action!

Looking forward to the new release on wordpress.org, so I can try it out. I'm using Fancybox, so I should be OK. Any idea when you will release?

**Comment by cristianraiber** [comment]
@ElectricFeet - we're trying to release today, but could be next Monday depending on how the last round of testing goes :)

**Comment by ElectricFeet** [comment]
@giucu91  @cristianraiber I just updated. It works a treat, thanks!
